subprojects/diagnostics/src/main/java/org/gradle/configuration/TaskDetailsModel.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configuration;
+
+import org.gradle.api.Task;
+import org.gradle.api.internal.tasks.options.OptionReader;
+import org.gradle.execution.TaskSelectionException;
+import org.gradle.execution.TaskSelector;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A configuration-friendly view of a task selection.
+ */
+public class TaskDetailsModel {
+    private final String taskPath;
+    private final List<TaskDetails> tasks;
+    private final TaskSelectionException failure;
+
+    private TaskDetailsModel(String taskPath, List<TaskDetails> tasks) {
+        this.taskPath = taskPath;
+        this.failure = null;
+        this.tasks = new ArrayList<>(tasks);
```

Comment:
🟡 It seems this defensive copy is unnecessary since the collection is created by this same class just a few lines below, what do you think?

Reply:
Doing the collection to sorted list mapping in the factory method now